src/adapters/mc/OsConfigResource.c

```diff
@@ -312,6 +312,44 @@ static MI_Result GetReportedObjectValueFromDevice(const char* who, MI_Context* c
     return miResult;
 }
 
+static char* GetReasonFromLog(const char* commandTemplate, char separator, const char* keyName, void* log)

```

Comment:
Why not build up the reasons "as you go" instead of searching thtough the logs?

Reply:
This feels like there will be alot of file reads/writes

Reply:
Thanks Captain! Let's continue (because we talked about this already) offline.